hoomd_polymers/base/molecule.py

```diff
@@ -0,0 +1,319 @@
+import itertools
+import random
+from typing import Union, Dict, List
+
+import mbuild as mb
+from gmso.external.convert_mbuild import from_mbuild
+from hoomd.md.force import Force as HForce
+from mbuild.lib.recipes import Polymer as mbPolymer
+
+from hoomd_polymers.base.base_types import FF_Types
+from hoomd_polymers.utils import check_return_iterable
+from .forcefield import find_xml_ff, apply_xml_ff, _validate_hoomd_ff
+
+
+class Molecule:
+    def __init__(self, n_mols, force_field: Union[Dict, List[HForce], str]=None, smiles=None, file=None, description=None,
+                 remove_hydrogens=False):
+        self.n_mols = check_return_iterable(n_mols)
+        self.force_field = force_field
+        self.smiles = smiles 
+        self.file = file 
+        self.description = description
+        self.remove_hydrogens = remove_hydrogens
+        self._mapping = None
+        self._mb_molecule = self._load()
+        self.gmso_molecule, self.topology_information = self._get_topology_information()
+        if self.force_field:
+            self._validate_force_field()
+        self._molecules = []
+        self._generate()
+
+
+
+    @property
+    def molecules(self):
+        """List of all instances of the molecule"""
+        return self._molecules
+    
+    @property
+    def mapping(self):
+        """Dictionary of particle index to bead mapping"""
+        return self._mapping
+
+    @mapping.setter
+    def mapping(self, mapping_array):
+        self._mapping = mapping_array
+
+    @property
+    def particle_types(self):
+        return self.topology_information["particle_types"]
+
+    @property
+    def pair_types(self):
+        return self.topology_information["pair_types"]
+
+    @property
+    def bond_types(self):
+        return self.topology_information["bond_types"]
+
+    @property
+    def angle_types(self):
+        return self.topology_information["angle_types"]
+
+    @property
+    def diherdal_types(self):
+        return self.topology_information["dihedral_types"]
+
+    @property
+    def improper_types(self):
+        return self.improper_types["improper_types"]
+
+    def _load(self):
+        if self.file and isinstance(self.file, str): # Loading from file takes precedent over SMILES 
+            return mb.load(self.file)
+        elif self.smiles and isinstance(self.smiles, str):
+            return mb.load(self.smiles, smiles=True)
+        else:
+            raise ValueError(
+                    "Unable to load from ",
+                    f"File: {self.file}",
+                    f"SMILES: {self.smiles}"
+            )
+
+    def _generate(self):
+        self._molecules.append(self._mb_molecule)
```

Comment:
Do we need a for loop here using `n_mols`? 

```
for i in range(self.n_mols):
    self._molecules.append(self._mb_molecule)
```

Reply:
Yeah, good catch! I'll fix it.